### PR TITLE
Daniil/rewrite-functional-iterators-into-for-loop-statement

### LINKED ISSUE #1952: Rewrite functional iterators into for loop statement
In go 1.23 [they accepted](https://tip.golang.org/doc/go1.23) functional iterators. We haven't used for loop statement because it wasn't clear what final interface would look like. 
One can find all functional iterators by searching usages of `iter.Seq` or `iter.Seq2` types.
Example:
```
eventsIt(func(res *spec.EventsResponse) bool {
	switch v := res.EventMessage.(type) {
	case *spec.EventsResponse_Event:
		events = append(events, v.Event)
		return true
	case *spec.EventsResponse_Fin:
		return false
	default:
		s.log.Warnw("Unexpected EventMessage from getEvents", "v", v)
		return false
	}
})
```
to be:
```
loop:
for res := range eventsIt {
	switch v := res.EventMessage.(type) {
	case *spec.EventsResponse_Event:
		events = append(events, v.Event)
	case *spec.EventsResponse_Fin:
                break loop // mind the loop label, because just break key word doesn't affect iteration
	default:
		s.log.Warnw("Unexpected EventMessage from getEvents", "v", v)
		break loop
	}
})
```
or
```
for res := range eventsIt {
	switch v := res.EventMessage.(type) {
	case *spec.EventsResponse_Event:
		events = append(events, v.Event)
                continue // as far as I remember continue works fine inside select
	case *spec.EventsResponse_Fin:
                // will break in the end
	default:
		s.log.Warnw("Unexpected EventMessage from getEvents", "v", v)
	}
        // comment why
	break
})
```

P.S. First you need to add `GOEXPERIMENT=rangefunc` env var. I suggest to put it on top of Makefile

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
fixes #1952

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 .DEFAULT_GOAL := help
 
+export GOEXPERIMENT=rangefunc
```
Comment (kirugan) [inline Makefile:3]: let's add comment on top `todo remove it after update to go 1.23`

File: p2p/sync.go [hunk 2]
```diff
@@ -409,26 +409,25 @@ func (s *syncService) genHeadersAndSigs(ctx context.Context, blockNumber uint64)
...
 			default:
 				s.log.Warnw("Unexpected HeaderMessage from getBlockHeaders", "v", v)
-				return false
+				break loop
 			}
 
 			select {
 			case <-ctx.Done():
-				return false
+				break loop
```
Comment (kirugan) [inline p2p/sync.go:427]: label was not necessary, but let's keep it as is

Comment (AnkushinDaniil) [reply p2p/sync.go:427]: [done](https://github.com/NethermindEth/juno/pull/1954/commits/3e10bbbb4d92e126b5da30f74ab415f41527032e)

File: Makefile (hunk 1 above)
Comment (AnkushinDaniil) [reply Makefile:4]: [done](https://github.com/NethermindEth/juno/pull/1954/commits/3e10bbbb4d92e126b5da30f74ab415f41527032e)

Comment (kirugan) [reply Makefile:4]: comments in Makefile starts with `#`

**Review by kirugan — APPROVED** [review]

**Review by derrix060 — CHANGES_REQUESTED** [review]
Failing on lint

**Comment by AnkushinDaniil** [comment]
Currently [golangci-lint does not support GOEXPERIMENT=rangefunc](https://github.com/golangci/golangci-lint/issues/4627), we have to wait for an update.

**Review by IronGauntlets — APPROVED** [review]

**Pull request merged**